File: app/src/main/java/com/uramonk/androidtemplateapp/viewmodel/NextFragmentViewModel.java
package com.uramonk.androidtemplateapp.viewmodel;

import com.trello.rxlifecycle.components.RxFragment;

import timber.log.Timber;

/**
 * Created by uramonk on 2016/07/04.
 */
public class NextFragmentViewModel extends BaseViewModel {
    private RxFragment fragment;

    public NextFragmentViewModel(RxFragment fragment) {
        super(fragment);
        this.fragment = fragment;
    }

    @Override
    protected void onCreate() {
        super.onCreate();
        Timber.d("onCreate");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Timber.d("onStart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Timber.d("onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Timber.d("onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Timber.d("onStop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Timber.d("onDestroy");
    }
}

File: app/src/main/java/com/uramonk/androidtemplateapp/viewmodel/NextFragmentViewModel.kt
package com.uramonk.androidtemplateapp.viewmodel

import com.trello.rxlifecycle.components.RxFragment

import timber.log.Timber

/**
 * Created by uramonk on 2016/07/04.
 */
class NextFragmentViewModel(private val fragment: RxFragment) : BaseViewModel(fragment) {

    override fun onCreate() {
        super.onCreate()
        Timber.d("onCreate")
    }

    override fun onStart() {
        super.onStart()
        Timber.d("onStart")
    }

    override fun onResume() {
        super.onResume()
        Timber.d("onResume")
    }

    override fun onPause() {
        super.onPause()
        Timber.d("onPause")
    }

    override fun onStop() {
        super.onStop()
        Timber.d("onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Timber.d("onDestroy")
    }
}
